cloud/docker/docker_network.py

```diff
@@ -0,0 +1,312 @@
+        cannot be deleted. Use the force option to disconnect all containers
+        and delete the network.
+      - <
+        "present" creates the network, if it does not already exist with the
+        specified parameters, and connects the list of containers provided via
+        the connected parameter. Containers not on the list will be
+        disconnected. An empty list will leave no containers connected to the
+        network. Use the incremental option to leave existing containers
+        connected. Use the force options to force re-creation of the network.
+    default: present
+    choices:
+      - absent
+      - present
+extends_documentation_fragment:
+    - docker
+authors:
+    - "Ben Keith (@keitwb)"
+requirements:
+    - "python >= 2.6"
+    - "docker-py >= 1.5.0"
+    - "The docker server >= 1.9.0"
+'''
+
+EXAMPLES = '''
+- name: Create a network
+  docker_network:
+    name: network_one
+
+- name: Remove all but selected list of containers
+  docker_network:
+    name: network_one
+    connected:
+      - container_a
+      - container_b
+      - container_c
+
+- name: Remove a single container
+  docker_network:
+    name: network_one
+    connected: "{{ fulllist|difference(['container_a']) }}"
+
+- name: Add a container to a network, leaving existing containers connected
+  docker_network:
+    name: network_one
+    connected:
+      - container_a
+    incremental: yes
+
+- name: Create a network with options
+  docker_network
+    name: network_two
+    driver_options:
+      com.docker.network.bridge.name: net2
+    ipam_options:
+      subnet: '172.3.26.0/16'
+      gateway: 172.3.26.1
+      iprange: '192.168.1.0/24'
+
+- name: Delete a network, disconnecting all containers
+  docker_network:
+    name: network_one
+    state: absent
+    force: yes
+'''
+
+RETURN = '''
+network:
+    description: Network inspection results for the affected network.
+    returned: success
+    type: complex
+    sample: {}
+'''
+
+
+from ansible.module_utils.docker_common import *
+
+try:
+    from docker import utils
+    from docker.utils.types import Ulimit
+except:
+    # missing docker-py handled in ansible.module_utils.docker
+    pass
+
+
+class TaskParameters(DockerBaseClass):
+    def __init__(self, client):
+        super(TaskParameters, self).__init__()
+        self.client = client
+
+        self.network_name = None
+        self.connected = None
+        self.driver = None
+        self.driver_options = None
+        self.ipam_driver = None
+        self.ipam_options = None
+        self.incremental = None
+        self.force = None
+
+        for key, value in client.module.params.items():
+            setattr(self, key, value)
+
+
+class DockerNetworkManager(object):
+
+    def __init__(self, client):
+        self.client = client
+        self.parameters = TaskParameters(client)
+        self.check_mode = self.client.check_mode
+        self.results = {'changed': False, 'actions': []}
```

Comment:
yeah, there's disagreement over how/when to return *actions*. For now, please only return it on check_mode True.